common/src/util/og.ts

```diff
@@ -0,0 +1,17 @@
+import { DOMAIN } from 'common/envs/constants'
+import { removeUndefinedProps } from 'common/util/object'
+
+export function buildOgUrl<P extends Record<string, string | undefined>>(
+  props: P,
+  endpoint: string
+) {
+  const generateUrlParams = (params: P) =>
+    new URLSearchParams(removeUndefinedProps(params) as any).toString()
```

Comment:
I don't see the use of making this anonymous function just so you can call it once below, instead of just writing
```typescript
const queryString = new URLSearchParams(params).toString()
```